src/rm/rm.rs

```diff
@@ -129,7 +129,7 @@ fn remove(files: Vec<String>, force: bool, interactive: InteractiveMode, one_fs:
         let filename = &filename[..];
         let file = Path::new(filename);
         if file.exists() {
-            if file.is_dir() {
+            if file.symlink_metadata().unwrap().is_dir() {
```

Comment:
You should report an error if `symlink_metadata()` fails rather than using `unwrap()`.